src/snowflake/snowpark/dataframe_transform_functions.py

```diff
@@ -57,6 +69,95 @@ def _validate_formatter_argument(self, fromatter):
         if not callable(fromatter):
             raise TypeError("formatter must be a callable function")
 
+    def _validate_and_extract_time_unit(
+        self, time_str: str, argument_name: str, allow_negative: bool = True
+    ) -> Tuple[int, str]:
+        if not time_str:
+            raise ValueError(f"{argument_name} must not be empty")
+
+        duration = int(time_str[:-1])
+        unit = time_str[-1].lower()
+
+        if not allow_negative and duration < 0:
+            raise ValueError(f"{argument_name} must not be negative.")
+
+        supported_units = ["s", "m", "h", "d", "w", "t", "y"]
+        if unit not in supported_units:
+            raise ValueError(
+                f"Unsupported unit '{unit}'. Supported units are '{supported_units}"
+            )
+
+        # Converting month unit to 'mm' for Snowpark
+        if unit == "t":
+            unit = "mm"
```

Comment:
I'm not familiar with Snowpark enough, so curious where does snowpark require `mm`?

Reply:
Thats the convention snowflake follows :)